src/test/java/org/sagebionetworks/bridge/spring/controllers/AuthenticationControllerTest.java

```diff
@@ -1165,7 +1182,125 @@ public void unconsentedReauthSetsMetrics() throws Exception {
+                .withRoles(ImmutableSet.of(DEVELOPER, ADMIN)).build());
+        doReturn(userSession).when(controller).getAuthenticatedSession();
+        
+        when(mockAccountDao.getStudyIdsForUser(SYNAPSE_USER_ID))
+            .thenReturn(ImmutableList.of("my-new-study"));
+        
+        Account account = Account.create();
+        AccountId accountId = AccountId.forSynapseUserId("my-new-study", SYNAPSE_USER_ID);
+        when(mockAccountDao.getAccount(accountId)).thenReturn(account);
+        
+        Study newStudy = Study.create();
+        newStudy.setIdentifier("my-new-study");
+        when(mockStudyService.getStudy("my-new-study")).thenReturn(newStudy);
+
+        UserSession session = new UserSession();
+        session.setSessionToken("new-session-token");
+        when(mockAuthService.getSessionFromAccount(eq(newStudy), any(), eq(account))).thenReturn(session);
+        
+        JsonNode node = controller.changeStudy();
+        assertEquals(node.get("sessionToken").textValue(), "new-session-token");
+
+        InOrder inOrder = Mockito.inOrder(mockAuthService, mockCacheProvider);
+        inOrder.verify(mockAuthService).signOut(userSession);
+        inOrder.verify(mockAuthService).getSessionFromAccount(eq(newStudy), any(), eq(account));
+        inOrder.verify(mockCacheProvider).setUserSession(session);
+    }
+    
+    @Test(expectedExceptions = UnauthorizedException.class, 
+            expectedExceptionsMessageRegExp = ".*" + STUDY_ACCESS_EXCEPTION_MSG + ".*")
+    public void changeStudyNotAuthorized() throws Exception {
+        mockRequestBody(mockRequest, new SignIn.Builder().withStudy("my-new-study").build());
+        doReturn(userSession).when(controller).getAuthenticatedSession();
+        
+        controller.changeStudy();
+    }
+
+    @Test(expectedExceptions = BadRequestException.class, 
+            expectedExceptionsMessageRegExp=".*Account has not been assigned a Synapse user ID.*")
+    public void changeStudyNotAssignedSynapseId() throws Exception {
+        mockRequestBody(mockRequest, new SignIn.Builder().withStudy("my-new-study").build());
+        userSession.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(DEVELOPER)).build());
+        doReturn(userSession).when(controller).getAuthenticatedSession();
+        
+        controller.changeStudy();
+    }
+    
+    @Test
+    public void changeStudySupportsCrossStudyAdmin() throws Exception {
+        mockRequestBody(mockRequest, new SignIn.Builder().withStudy("my-new-study").build());
+        userSession.setParticipant(new StudyParticipant.Builder().withSynapseUserId(SYNAPSE_USER_ID)
+                .withRoles(ImmutableSet.of(ADMIN)).build());
+        doReturn(userSession).when(controller).getAuthenticatedSession();
+        
+        when(mockAccountDao.getStudyIdsForUser(SYNAPSE_USER_ID))
+            .thenReturn(ImmutableList.of("my-new-study", TEST_STUDY_IDENTIFIER));
```

Comment:
In practice, admin accounts would only have TEST_STUDY_IDENTIFIER show up in their study list.

Reply:
This is true. I've removed the other ID which is a stronger test (we specifically don't want the list to match the target study).